scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,788 @@
+                                                        width: width,
+                                                        height: height,
+                                                        background: 'unset',
+                                                        position: 'absolute',
+                                                    }}
+                                                />
+                                            )}
+                                        </div>
+                                        <VideoEditorTools
+                                            wrapperRef={this.setVideoToolsSize}
+                                            playPause={this.playPause}
+                                            onRotate={this.handleRotate}
+                                            onCrop={this.handleToggleCrop}
+                                            onQualityChange={this.handleQualityChange}
+                                            cropEnabled={this.state.cropEnabled}
+                                            videoDegree={this.state.transformations.degree}
+                                            videoPlaying={this.state.playing}
+                                            videoQuality={this.state.transformations.quality}
+                                            videoHeadline={this.props.article.headline}
+                                            videoResolution={Math.min(
+                                                videoRef?.videoWidth ?? 0,
+                                                videoRef?.videoHeight ?? 0,
+                                            )}
+                                            gettext={gettext}
+                                            getClass={getClass}
+                                        />
+                                    </div>
+                                </div>
+                                <div className="sd-photo-preview__thumb-strip sd-photo-preview__thumb-strip--video">
+                                    {videoRef && (
+                                        <>
+                                            <VideoEditorThumbnail
+                                                video={videoRef}
+                                                article={this.props.article}
+                                                etag={this.state.etag}
+                                                renditions={this.state.renditions}
+                                                onToggleLoading={this.handleToggleLoading}
+                                                onSave={(renditions: IArticle['renditions'], etag: string) =>
+                                                    // update preview thumbnail will change etag
+                                                    this.setState({renditions: renditions, etag: etag})
+                                                }
```

Comment:
Upload thumbnail will change article etag, we think it's possible to do it on the server, but need some hacks, so we think it's better to manually sync it on the client